doozer/doozerlib/backend/konflux_image_builder.py

```diff
@@ -515,7 +515,7 @@ async def _get_for_arch(arch):
                             # get the source rpm
                             source_rpm = purl.qualifiers.get("upstream", None)
                             if source_rpm:
-                                source_rpms.add(source_rpm.rstrip(".src.rpm"))
+                                source_rpms.add(source_rpm.split(".src.rpm")[0])
```

Comment:
Are you looking for `removesuffix`? https://docs.python.org/3/library/stdtypes.html#str.removesuffix
```suggestion
                                source_rpms.add(source_rpm. removesuffix(".src.rpm"))
```